src/reducers/versions.tsx

```diff
@@ -0,0 +1,218 @@
+  entries: VersionEntry[];
+  id: VersionId;
+  reviewed: string;
+  version: string;
+};
+
+export const actions = {
+  loadVersionFile: createAction('LOAD_VERSION_FILE', (resolve) => {
+    return (payload: { path: string; version: ExternalVersion }) =>
+      resolve(payload);
+  }),
+  loadVersionInfo: createAction('LOAD_VERSION_INFO', (resolve) => {
+    return (payload: { version: ExternalVersion }) => resolve(payload);
+  }),
+};
+
+export type VersionsState = {
+  versionInfo: {
+    [versionId: number]: Version;
+  };
+  versionFiles: {
+    [versionId: number]: {
+      [path: string]: VersionFile;
+    };
+  };
+};
+
+export const initialState: VersionsState = {
+  versionInfo: {},
+  versionFiles: {},
+};
+
+export const getVersionEntryType = (entry: ExternalVersionEntry) => {
+  if (entry.directory) {
+    return VersionEntryType.directory;
+  }
+  if (entry.binary === 'image') {
+    return VersionEntryType.image;
+  }
+  if (entry.binary) {
+    return VersionEntryType.binary;
+  }
+  return VersionEntryType.text;
+};
+
+export const createInternalVersionFile = (
+  file: ExternalVersionFile,
+): VersionFile => {
+  return {
+    content: file.content,
+    created: file.created,
+    id: file.id,
+    size: file.size,
+  };
+};
+
+export const createInternalVersion = (version: ExternalVersion): Version => {
+  return {
+    entries: Object.keys(version.file.entries).map((nodeName) => {
+      const entry = version.file.entries[nodeName];
+      return {
+        type: getVersionEntryType(entry),
+        depth: entry.depth,
+        filename: entry.filename,
+        modified: entry.modified,
+        path: entry.path,
+      };
+    }),
+    id: version.id,
+    reviewed: version.reviewed,
+    version: version.version,
+  };
+};
+
+export const getVersionFile = (
+  path: string,
+  versions: VersionsState,
+  versionId: VersionId,
+) => {
+  const filesForVersion = versions.versionFiles[versionId];
+  return filesForVersion ? filesForVersion[path] : undefined;
+};
+
+export const getVersionInfo = (
+  versions: VersionsState,
+  versionId: VersionId,
+) => {
+  return versions.versionInfo[versionId];
+};
+
+const reducer: Reducer<VersionsState, ActionType<typeof actions>> = (
+  state = initialState,
+  action,
+): VersionsState => {
+  switch (action.type) {
+    case getType(actions.loadVersionInfo): {
+      const { version } = action.payload;
+      return {
+        ...state,
+        versionInfo: {
+          ...state.versionInfo,
+          [version.id]: createInternalVersion(version),
+        },
+      };
+    }
+    case getType(actions.loadVersionFile): {
+      const { path, version } = action.payload;
+      return {
+        ...state,
+        versionFiles: {
+          ...state.versionFiles,
+          [version.id]: {
+            ...state.versionFiles[version.id],
+            [path]: createInternalVersionFile(version.file),
+          },
+        },
+      };
+    }
+    default:
+      return state;
```

Comment:
this is not covered by a test case 🤓

Reply:
Yes, I noticed this. I believe this is because we decided to remove the tests for reducers that simply check that `initialState` is returned when an invalid action is passed. Maybe we want to add something like that back in so that we can have 100% coverage? @kumar303 what do you think?

Reply:
ah right, I remember a comment mentioning that such test cases had little value. I do agree. We could also add a comment to ignore the coverage.